boot/zephyr/CMakeLists.txt

```diff
@@ -213,5 +213,19 @@ if(NOT CONFIG_BOOT_SIGNATURE_KEY_FILE STREQUAL "")
     DEPENDS ${KEY_FILE}
     )
   zephyr_library_sources(${GENERATED_PUBKEY})
+
+  set(GENERATED_PRIVKEY ${ZEPHYR_BINARY_DIR}/autogen-privkey.c)
```

Comment:
I'll be honest, I don't want this feature. It isn't going to do anything for me, other than make development slightly less convenient, when, after I make clean, I have a new keypair being used.

At a minimum, can we make this a configurable option? The default could be to use the hard-coded key, and with an option, it will generate one for that build. Really, to do secure key management, the private key would need to come from an HSM, which is something on our longer-term task list.

Reply:
Thinking more about this, this would address #1094, which causes us to get periodic security vulnerability reports, because of the in-repo keys.